src/Type/Accessory/AccessoryNonEmptyStringType.php

```diff
@@ -262,6 +264,12 @@ public function isScalar(): TrinaryLogic
 
 	public function looseCompare(Type $type, PhpVersion $phpVersion): BooleanType
 	{
+		$looseFalse = new NullType();
+
+		if ($looseFalse->isSuperTypeOf($type)->yes()) {
```

Comment:
We can be sure it's going to be always false with:

* `''`
* `false`

Reply:
I see that this test passes but I have no idea how:

```
assertType('false', $nonEmptyString == $emptyStr);
```

Reply:
> I see that this test passes but I have no idea how

it works because it is resolved via the "identical" type in https://github.com/phpstan/phpstan-src/blob/d01b91999311fbcdc091e105ac46df536dcc5648/src/Reflection/InitializerExprTypeResolver.php#L1330-L1337

it will start failing as expected, after https://github.com/phpstan/phpstan-src/pull/2228 is merged :)

Reply:
`'0' == false` is `true`